src/main.rs

```diff
@@ -1290,6 +1290,29 @@ fn dbus_attiny_command(
     conn: &mut DuplexConn,
     command: u8,
     value: Option<u8>,
+) -> Result<u8, &'static str> {
+    let max_attempts = 10;
+    let retry_delay = std::time::Duration::from_secs(2);
+
+    for attempt in 1..=max_attempts {
+        match dbus_attiny_command_attempt(conn, command, value) {
+            Ok(result) => return Ok(result),
+            Err(e) => {
+                if attempt >= max_attempts {
```

Comment:
Very small nit: `attempt` can never be greater than `max_attempts` as you've written the range as `1..=max_attempts`, so `==` equality is probably what you want.